### DISCUSSION TITLE
Send Uploaded File Content to Rest API

### DISCUSSION THREAD

**Discussion by HeerlijkeHeer** [Budibase/budibase#14118] [Help]
Hello Budibase community

I have a REST API that accepts the (binary) contents of an Excel document and uploads that data to a SQL database. The API performs a lot of parsing and enrichment, so it's not a straight-up upload.
In addition to editing rows of data, I want the users of my Budibase app to be able to perform a "bulk upload" by uploading an Excel document with their data. How could I achieve this in Budibase? 

 I've created a form with a `Single Attachment` component and a submit `Button`. The problem I now face is that I have no idea how I can:
1. Retrieve the (binary) contents of the attachment.
2. Send that content to the REST API. I.e. I have no idea how to use the `files` property when building API queries in Budibase.

Thank you for any help you can offer!

**Comment by mikesealey** [comment]
Hey @HeerlijkeHeer 

You can certainly upload a file as a single-attachment and send it to an API, but Budibase doesn't currently support (or have plans to support) retrieving binary data from an attachment. 

If that doesn't immediately quash your feature, head to the documentation on [REST queries](https://docs.budibase.com/docs/rest-queries).

**Comment by papka1974** [comment]
I bump into the same problem. I have some flows in NODE-RED, where EXCEL files should be posted, then they processed there and data inserted into database (Firebird). I'm new in Budibase and trying to use it as frontend, so didn't found any other way to do this as just create a "fake" Budibase table with attachments field, to be able to use attachments component, added a button, which using api sending attachments component state to NODE-RED, and NODE-RED reading attached files itself from MINIO. I opened MINIO port for that.
Don't know is it good idea and will bring you anything, but decided to share.
